### DISCUSSION TITLE
Enhancing Crazyflie Swarm Demo for Loco-Positioning System and Firmware-Based Control

### DISCUSSION THREAD

**Discussion by amitx64** [bitcraze/discussions#1150] [Q&A]
Hello Crazyflie community,

I'm excited to explore the possibilities of taking the existing swarm demo on Crazyflie to the next level. Currently, the swarm demo is demonstrated with the Lighthouse positioning system, but I'm interested in adapting it for use with the loco-positioning system. Additionally, I want to extend the functionality by introducing firmware-based control.

### Goals:

1. Loco-Positioning System Compatibility: I aim to adapt the swarm demo to work seamlessly with the loco-positioning system. This involves tweaking parameters and configurations to ensure accurate and reliable positioning.

2. Firmware-Based Control: Instead of relying solely on Python scripts for controlling the swarm, I want to implement firmware-based control. This means the Crazyflie should be capable of initiating swarming sequences either after an external trigger (e.g., RC Transmitter input) or autonomously after a specified time delay (e.g., 10 seconds).

### Discussion Points:

- Has anyone attempted to use the loco-positioning system with the swarm demo? Any insights or challenges encountered?
- Are there specific firmware hooks or APIs that can be leveraged to implement firmware-based control of the swarm?
- What considerations should be taken into account when adapting the existing swarm demo for loco-positioning?
- Any advice or guidance from those experienced with Crazyflie firmware development would be highly appreciated.

Let's collaborate and brainstorm ideas to enhance the Crazyflie swarm demo for a broader range of positioning systems and introduce advanced firmware-based control features. Your expertise and insights are invaluable in making this project a success!

**Comment by StewenceNewton** [comment]
Hi, My name is newton I have been working on these loco positioning systems along with the Crazyflie and these are responding to the code properly but the problem is with the flight stability. These are not staying in the particular position moving up and down  not able to hold in the position, I tried with the TWR, TdoA2 and TdoA3 mode. Can you help on this thing and I have used 8 loco nodes around 3mx5mx2m (XxYxZ) space. While controlling with cfclient also it is not able to stay in the single position and i'm attaching the video reference google drive link 

https://drive.google.com/file/d/1KvXaBkwejIIUrZCxXykX8o09dRPLcJMd/view?usp=sharing

**Reply by knmcguire** [reply]
Hi @StewenceNewton. Could you start a new discussion thread for this please if you haven't already?

**Reply by StewenceNewton** [reply]
Thank you! Ok i'll start the new discussion

**Reply by amitx64** [reply]
Hi @StewenceNewton, I am also working with LPS. However, I only have one Crazyflie with me, and it's hovering fine. If possible, could you please share what you're trying to accomplish and how you set up your system for swarming?

**Reply by StewenceNewton** [reply]
@amitx64 With one crazyflie using TWR mode it's hovering with good accuracy. While doing swarming under TDoA2 mode then the inaccuracy is occurring with increasing number of crazyflies

**Reply by amitx64** [reply]
@StewenceNewton I did a test of Crazyflie with autonomous flight where the Crazyflie followed a pre-fed trajectory of a figure 8 and then landed, and it flew fine. I think you placed your Loco node near walls or metal objects. Please avoid that.

**Reply by michaelcheng8** [reply]
@StewenceNewton this is a bit late to your reply, but have you been able to increase positioning accuracy while using TDoA2 or 3?
